graphql_compiler/macros/__init__.py

```diff
@@ -88,14 +97,23 @@ def register_macro_edge(macro_registry, macro_edge_graphql, macro_edge_args):
 def get_schema_with_macros(macro_registry):
     """Get a new GraphQLSchema with fields where macro edges can be used.
 
-    Specification:
+    Preconditions:
+    1. No macro in the registry has the same name as a field on the vertex where it applies,
+       or any interfaces it implements.
+    2. Members of a union type do not have outgoing macros with the same name.
```

Comment:
nit: an union

Reply:
I believe that `a union` might actually be correct, though it's definitely tricky and I had to look it up to be sure: https://english.stackexchange.com/questions/266309/why-is-union-an-exception-to-the-a-an-rule

Reply:
why?